Tests/SwiftParserTest/TypeTests.swift

```diff
@@ -304,4 +304,23 @@ final class TypeTests: ParserTestCase {
       ]
     )
   }
+
+  func testTypedThrows() {
+    assertParse(
+      """
+      { () throws(PosixError) -> Void in }
+      """,
+      experimentalFeatures: [.typedThrows]
+    )
+
+    assertParse(
+      "typealias T = () throws(PosixError) -> Void",
+      experimentalFeatures: [.typedThrows]
+    )
+
+    assertParse(
+      "[() throws(PosixError) -> Void]()",
+      experimentalFeatures: [.typedThrows]
+    )
+  }
```

Comment:
I think the following test cases would be interesting as well. I have not filled in the expected diagnostics here but we should make sure that they make sense

### Function parsing.
```swift
assertParse(
  "func test() throws(MyError) async {}"
)
assertParse(
  "func test() throws async(MyError) {}"
)
assertParse(
  "func test() try(MyError) async {}"
)
assertParse(
  "func test() try async(MyError) {}"
)
assertParse(
  "func test() throws(MyError) await {}"
)
assertParse(
  "func test() throws await(MyError) {}"
)
assertParse(
  "func test() try(MyError) await {}"
)
assertParse(
  "func test() try await(MyError) {}"
)
assertParse(
  "func test() async(MyError) {}"
)
assertParse(
  "func test() await(MyError) {}"
)
assertParse(
  "func test() try(MyError) {}"
)
assertParse(
  "func test() throws(MyError) {}"
)
assertParse(
  "func test() throws(MyError)async {}" // no space between closing parenthesis and `async`
)
```

### `isAtFunctionTypeArrow` disambiguation
```swift
assertParse(
  "[() throws(MyError) async -> Void]()"
)
assertParse(
  "[() throws async(MyError) -> Void]()"
)
assertParse(
  "[() try(MyError) async -> Void]()"
)
assertParse(
  "[() try async(MyError) -> Void]()"
)
assertParse(
  "[() throws(MyError) await -> Void]()"
)
assertParse(
  "[() throws await(MyError) -> Void]()"
)
assertParse(
  "[() try(MyError) await -> Void]()"
)
assertParse(
  "[() try await(MyError) -> Void]()"
)
assertParse(
  "[() async(MyError) -> Void]()"
)
assertParse(
  "[() await(MyError) -> Void]()"
)
assertParse(
  "[() try(MyError) -> Void]()"
)
assertParse(
  "[() throws(MyError) -> Void]()"
)
```

Reply:
Great test ideas, thanks!